onnxruntime/contrib_ops/webgpu/bert/bias_add.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/webgpu/shader_helper.h"
+#include "core/providers/webgpu/webgpu_supported_types.h"
+#include "contrib_ops/webgpu/bert/bias_add.h"
+#include "contrib_ops/webgpu/webgpu_contrib_kernels.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace webgpu {
+
+ONNX_OPERATOR_KERNEL_EX(
+    BiasAdd,
+    kMSDomain,
+    1,
+    kWebGpuExecutionProvider,
+    (*KernelDefBuilder::Create())
+        .TypeConstraint("T", WebGpuSupportedFloatTypes()),
+    BiasAdd);
+
+Status BiasAddProgram::GenerateShaderCode(ShaderHelper& shader) const {
+  const ShaderVariableHelper& input = shader.AddInput("input");
+  const ShaderVariableHelper& bias = shader.AddInput("bias");
+  const ShaderVariableHelper& residual = shader.AddInput("residual");
+  const ShaderVariableHelper& output = shader.AddOutput("output");
+
+  shader.MainFunctionBody() << shader.GuardAgainstOutOfBoundsWorkgroupSizes("uniforms.output_size")
+                            << "const channels = (" << channels_ << "/ 4)u;\n"
```

Comment:
Does it need to be a uniforms variable so that same shader code works for different possible values of channel? It is not in the hint. `AddInput` does not specify program dependence on input/output shape.

Reply:
I am working off the [reference implementation in jsep](https://github.com/microsoft/onnxruntime/blob/main/js/web/lib/wasm/jsep/webgpu/ops/bias-add.ts) in which channels is not a uniform variable and there is no cache hint. Additionally, there is no dependence on input/output shape. Only on output size for which there is precedent from other webgpu ep ops to store as a uniform variable.